src/components/canrawview/uniquefiltermodel.cpp

```diff
@@ -0,0 +1,52 @@
+#include "uniquefiltermodel.h"
+
+UniqueFilterModel::UniqueFilterModel(QObject* parent)
+    : QSortFilterProxyModel(parent)
+{
+}
+
+void UniqueFilterModel::updateFilter(int frameID, double time, QString direction)
+{
+    QPair< int, QString> value(frameID, direction);
+
+    if ((!uniques.contains(value)) || (time > uniques[value]))
+    {
+        uniques[value] = time;
```

Comment:
If this works as `operator[]` in `std::map`, then I would propose leaving `value` non-const and use `uniques[std::move(value)] = time;`